src/components/App.tsx

```diff
@@ -601,86 +546,44 @@ class App extends React.Component<ExcalidrawProps, AppState> {
       );
     }
 
-    const searchParams = new URLSearchParams(window.location.search);
-    const id = searchParams.get("id");
-    const jsonMatch = window.location.hash.match(
-      /^#json=([0-9]+),([a-zA-Z0-9_-]+)$/,
-    );
-
     if (!this.state.isLoading) {
       this.setState({ isLoading: true });
     }
 
-    let scene = await loadScene(null, null, this.props.initialData);
-
-    let isCollaborationScene = !!getCollaborationLinkData(window.location.href);
-    const isExternalScene = !!(id || jsonMatch || isCollaborationScene);
-
-    if (isExternalScene) {
-      if (
-        this.shouldForceLoadScene(scene) ||
-        window.confirm(t("alerts.loadSceneOverridePrompt"))
-      ) {
-        // Backwards compatibility with legacy url format
-        if (id) {
-          scene = await loadScene(id, null, this.props.initialData);
-        } else if (jsonMatch) {
-          scene = await loadScene(
-            jsonMatch[1],
-            jsonMatch[2],
-            this.props.initialData,
-          );
-        }
-        if (!isCollaborationScene) {
-          window.history.replaceState({}, "Excalidraw", window.location.origin);
-        }
-      } else {
-        // https://github.com/excalidraw/excalidraw/issues/1919
-        if (document.hidden) {
-          window.addEventListener("focus", () => this.initializeScene(), {
-            once: true,
-          });
-          return;
-        }
+    let data = null;
+    try {
+      data = await this.props.initialData;
+    } catch (e) {}
```

Comment:
We should at the very least `console.error` this, but I'm thinking we should maybe let it fail?

Reply:
Why break the App if due to some reason `initialData` fails ?

Reply:
Something will break and it may be better to let the host app know (and handle it itself) rather than silently swallow it, no?

Reply:
Let me put it this way: if the host app has an `initialData` getter, a sync error will break the app, while an async error will be silently swallowed. That inconsistency is weird and not justified.

Reply:
of course, we should have `console.error` but shouldn't we still handle it in our code, and since we are logging it so not worry about how the host is going to handle it?

Reply:
Ideally we should think about host apps too, not just ours.

Reply:
I get your point that's why we are logging it so the host can also handle it accordingly. 
IMO we should always do error handling in our code. 
Having said that I don't have a strong opinion against having it fail so we can let it fail and iterate over it later if needed.